akka-cluster/src/main/scala/akka/cluster/CrossDcClusterHeartbeat.scala

```diff
@@ -0,0 +1,321 @@
+    else false
+  }
+
+  /** Idempotent, become active if this node is n-th oldest and should monitor other nodes */
+  private def becomeActiveIfResponsibleForHeartbeat(): Unit = {
+    if (!activelyMonitoring && selfIsResponsibleForCrossDcHeartbeat()) {
+      if (verboseHeartbeat) log.debug("Becoming ACTIVE (for DC: {}), monitoring other DCs oldest nodes.", selfDataCenter)
+      activelyMonitoring = true
+
+      context.become(active orElse introspecting)
+    } else if (!activelyMonitoring)
+      if (verboseHeartbeat) log.info("Remaining DORMANT; others in {} handle monitoring for me ({})", selfDataCenter)
+  }
+
+}
+
+/** INTERNAL API */
+@InternalApi
+private[akka] object CrossDcHeartbeatSender {
+
+  // -- messages intended only for local messaging during testing -- 
+  sealed trait InspectionCommand extends NoSerializationVerificationNeeded
+  final case class ReportStatus()
+
+  sealed trait StatusReport extends NoSerializationVerificationNeeded
+  sealed trait MonitoringStateReport extends StatusReport
+  final case class MonitoringActive(state: CrossDcHeartbeatingState) extends MonitoringStateReport
+  final case class MonitoringDormant() extends MonitoringStateReport
+  // -- end of messages intended only for local messaging during testing -- 
+}
+
+/** INTERNAL API */
+@InternalApi
+private[cluster] final case class CrossDcHeartbeatingState(
+  failureDetector:         FailureDetectorRegistry[Address],
+  nrOfMonitoredNodesPerDc: Int,
+  state:                   Map[ClusterSettings.DataCenter, SortedSet[Member]]) {
+  import CrossDcHeartbeatingState._
+
+  /**
+   * Decides if `self` node should become active and monitor other nodes with heartbeats.
+   * Only the `nrOfMonitoredNodesPerDc`-oldest nodes in each DC fulfil this role.
+   */
+  def shouldActivelyMonitorNodes(selfDc: ClusterSettings.DataCenter, selfAddress: UniqueAddress): Boolean = {
+    /** Since we need ordering of oldests guaranteed, we must only look at Up (or Leaving, Exiting...) nodes */
+    def atLeastInUpState(m: Member): Boolean =
+      m.status != MemberStatus.WeaklyUp && m.status != MemberStatus.Joining
+
+    val selfDcNeighbours: SortedSet[Member] = state.getOrElse(
+      selfDc,
+      throw new IllegalStateException(s"Self DataCenter was [$selfDc] however no such entries in ${Logging.simpleName(getClass)} present. " +
+        s"This is very likely a bug, please turn on debug logging and open an issue about it."))
+    val selfDcOldOnes = selfDcNeighbours.filter(atLeastInUpState).take(nrOfMonitoredNodesPerDc)
+
+    // if this node is part of the "n oldest nodes" it should indeed monitor other nodes:
+    val shouldMonitorActively = selfDcOldOnes.exists(_.uniqueAddress == selfAddress)
+    shouldMonitorActively
+  }
+
+  def addMember(m: Member): CrossDcHeartbeatingState = {
+    val dc = m.dataCenter
+    def addMember0(member: Member): CrossDcHeartbeatingState = {
+      val updatedMembers = state.getOrElse(dc, emptyMembersSortedSet) + m
```

Comment:
This is a bit sneaky, since we do an removeMember before calling addMember0; 
Can it remain so or do you think the filter here would be nicer?

There is a slight semantic difference I think in timing, since the removeMember actually removes from the failure detector. And if we filter here it would not. The difference is I guess on the timeout of triggering in the FD...

I think the filtering here, since the address in that case would be the exact same in the failure detector, would perhaps yield the more expected result?